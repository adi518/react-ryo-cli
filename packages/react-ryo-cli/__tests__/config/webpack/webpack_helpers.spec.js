const slash = require("slash");

const { SCRIPTS } = require("../../../lib/constants");

const {
  getOutputPath,
  overrideWebpackConfig
} = require("../../../config/webpack/webpack_helpers.js");

console.log(getOutputPath(undefined, { cwd: "/app" }));

const commonMockArgs = {
  paths: {},
  cliOptions: {},
  source: {
    output: {},
    entry: ["index.js"],
    resolve: { plugins: [] }
  },
  getOutputPath: () => slash(getOutputPath(undefined, { cwd: "/app" }))
};

const getMockArgs = options => ({ ...commonMockArgs, ...options });

jest.mock("../../../lib/helpers", () => ({
  ...jest.requireActual("../../../lib/helpers"),
  getLibraryName: () => "react-core"
}));

describe("overrideWebpackConfig", () => {
  test("with build script", () => {
    const { source, ...options } = getMockArgs({ script: SCRIPTS.BUILD });
    const result = overrideWebpackConfig(source, options);

    expect(result).toMatchSnapshot();
  });

  test("with build:development script", () => {
    const { source, ...options } = getMockArgs({
      script: SCRIPTS.BUILD_DEVELOPMENT
    });
    const result = overrideWebpackConfig(source, options);

    expect(result).toMatchSnapshot();
  });

  test("with build:package script", () => {
    const { source, ...options } = getMockArgs({
      script: SCRIPTS.BUILD_PACKAGE,
      argv: { name: "react-core" }
    });
    const result = overrideWebpackConfig(source, options);

    expect(result).toMatchSnapshot();
  });

  test("with build:package:development script", () => {
    const { source, ...options } = getMockArgs({
      script: SCRIPTS.BUILD_PACKAGE_DEVELOPMENT,
      argv: { name: "react-core" }
    });
    const result = overrideWebpackConfig(source, options);

    expect(result).toMatchSnapshot();
  });

  test("getOutputPath", () => {
    expect(slash(getOutputPath(undefined, { cwd: "/app" }))).toMatchSnapshot();
  });
});
