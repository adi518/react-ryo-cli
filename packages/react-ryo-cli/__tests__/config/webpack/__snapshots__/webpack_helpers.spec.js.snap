// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extendWebpack extendCssLoaderOptions 1`] = `
Object {
  "modules": Object {
    "localIdentName": "kn-[name]__[local]___[hash:base64:5]",
  },
}
`;

exports[`extendWebpack getOutputPath 1`] = `"/app/out"`;

exports[`extendWebpack with build (development) script 1`] = `
Object {
  "extendedPaths": Object {
    "appBuild": "/app/out",
  },
  "extendedWebpackConfig": Object {
    "devtool": "eval-source-map",
    "entry": Array [
      "./node_modules/react-build/config/babel/babel_polyfill.js",
      "index.js",
    ],
    "externals": Object {},
    "mode": "development",
    "output": Object {
      "path": "/app/out",
    },
  },
}
`;

exports[`extendWebpack with build:package script 1`] = `
Object {
  "extendedPaths": Object {},
  "extendedWebpackConfig": Object {
    "devtool": "eval-source-map",
    "entry": Array [
      "index.js",
    ],
    "externals": Object {
      "react": "react",
      "react-dom": "react-dom",
      "styled-components": "styled-components",
    },
    "mode": "development",
    "output": Object {
      "library": "react-core",
      "libraryTarget": "umd",
      "path": "/app/out",
    },
  },
}
`;

exports[`extendWebpack with build:package:production script 1`] = `
Object {
  "extendedPaths": Object {},
  "extendedWebpackConfig": Object {
    "entry": Array [
      "index.js",
    ],
    "externals": Object {
      "react": "react",
      "react-dom": "react-dom",
      "styled-components": "styled-components",
    },
    "output": Object {
      "library": "react-core",
      "libraryTarget": "umd",
      "path": "/app/out",
    },
  },
}
`;

exports[`extendWebpack with build:production script 1`] = `
Object {
  "extendedPaths": Object {
    "appBuild": "/app/out",
  },
  "extendedWebpackConfig": Object {
    "entry": Array [
      "./node_modules/react-build/config/babel/babel_polyfill.js",
      "index.js",
    ],
    "externals": Object {},
    "output": Object {
      "path": "/app/out",
    },
  },
}
`;
